src/lib.rs

```diff
@@ -199,8 +229,83 @@ impl Contract {
         notify::notify_mentions(desc.as_str(), id);
     }
 
+    #[payable]
+    pub fn add_proposal(&mut self, body: VersionedProposalBody, labels: HashSet<String>) -> Promise {
+        near_sdk::log!("add_proposal");
+        let id = self.proposals.len();
```

Comment:
Let's convert it once here and don't `try_into()` it below.

```suggestion
        let id: ProposalId = self.proposals.len().try_into().unwrap();
```